### DISCUSSION TITLE
The Amber Pages directory

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#346] [Ideas]
An attempt at a witty reference to amber CRTs and the old White/Yellow pages for phone directories, I intend to put together a directory of nodes (not peers) in an index. While I could catch announces and list them dynamically, I'd like feedback on if the directory should be opt-in only. 

It's barely useful, as any given RNS instance will collect all the nodes in ~24 hours on the outside, but I think it might be useful for just starting up or if someone forgets to save nodes. At the very least it can help relieve the "I've been running NomadNet for an hour and don't see anyone! Did I mess up?" issues which, frankly, I had starting up as well. 

It wouldn't be an authoritative listing, after all we don't want that, but rather a service to retain addresses for new users. Combined with a TCP server and a published LXMF/node address, it could get a new user a toehold into using the network.

Does anyone have any feedback on the Announce scraping versus an opt-in system? I would argue an announce from a node is explicitly wishing to be found and listed in path listings, but I'd like feedback.

I am also aware there's already a page listing node announces, but I'm looking for all opinions before starting any real work.

**Comment by markqvist** [comment]
Heh, The Amber Pages is a great name :)

I think it'd be very useful, especially as a place to start for people new to the entire system. A nice entry point. Also really useful when you don't have the oppertunity to let your own node run constantly, but connect occasionally.

I think the general spirit is that anyone can and should link to anything that is accessible, so personally I feel no need for something like this to be opt-in, but of course, if anyone doesn't want to be listed, I'd respect that as well. But an open and permissionless approach is what I personally find most productive.

**Comment by faragher** [comment]
First implementation is up on the Between the Borders node. It collects all nomadnetwork.node announces.

I'm having some issues with daemonization and I had to restart the NomadNet client to get it to work for some odd reason, but it's technically sound.
